cpp/src/interop/from_arrow_device.cu

```diff
@@ -88,6 +93,309 @@ data_type arrow_to_cudf_type(const ArrowSchemaView* arrow_view)
+  if (has_nulls) {
+    auto out_mask = detail::copy_bitmask(static_cast<bitmask_type*>(get_mask_buffer(input)->data()),
+                                         input->offset,
+                                         input->offset + input->length,
+                                         stream,
+                                         mr);
+
+    out_col->set_null_mask(std::move(out_mask), input->null_count);
+  }
+
+  return out_col;
+}
+
+template <>
+std::unique_ptr<column> dispatch_copy_from_arrow_device::operator()<cudf::string_view>(
+  ArrowSchemaView* schema, ArrowArray const* input, data_type type, bool skip_mask)
+{
+  if (input->length == 0) { return make_empty_column(type_id::STRING); }
+
+  const void* offset_buffers[2] = {nullptr, input->buffers[fixed_width_data_buffer_idx]};
+  ArrowArray offsets_array      = {
+         .length     = input->offset + input->length + 1,
+         .null_count = 0,
+         .offset     = 0,
+         .n_buffers  = 2,
+         .n_children = 0,
+         .buffers    = offset_buffers,
+  };
+
+  size_type const char_data_length =
+    reinterpret_cast<int32_t const*>(offset_buffers[1])[input->length + input->offset];
+  const void* char_buffers[2] = {nullptr, input->buffers[2]};
+  ArrowArray char_array       = {
+          .length     = char_data_length,
+          .null_count = 0,
+          .offset     = 0,
+          .n_buffers  = 2,
+          .n_children = 0,
+          .buffers    = char_buffers,
+  };
+
+  nanoarrow::UniqueSchema offset_schema;
+  NANOARROW_THROW_NOT_OK(ArrowSchemaInitFromType(offset_schema.get(), NANOARROW_TYPE_INT32));
+
+  nanoarrow::UniqueSchema char_data_schema;
+  NANOARROW_THROW_NOT_OK(ArrowSchemaInitFromType(char_data_schema.get(), NANOARROW_TYPE_INT8));
+
+  ArrowSchemaView view;
+  NANOARROW_THROW_NOT_OK(ArrowSchemaViewInit(&view, offset_schema.get(), nullptr));
+  auto offsets_column =
+    this->operator()<int32_t>(&view, &offsets_array, data_type(type_id::INT32), true);
+  NANOARROW_THROW_NOT_OK(ArrowSchemaViewInit(&view, char_data_schema.get(), nullptr));
+  auto chars_column = this->operator()<int8_t>(&view, &char_array, data_type(type_id::INT8), true);
+
+  auto const num_rows = offsets_column->size() - 1;
+  auto out_col        = make_strings_column(num_rows,
+                                     std::move(offsets_column),
+                                     std::move(chars_column->release().data.release()[0]),
+                                     input->null_count,
+                                     std::move(*get_mask_buffer(input)));
+
+  return input->offset == 0
+           ? std::move(out_col)
+           : std::make_unique<column>(
+               cudf::detail::slice(out_col->view(),
+                                   static_cast<size_type>(input->offset),
+                                   static_cast<size_type>(input->offset + input->length),
+                                   stream),
+               stream,
+               mr);
```

Comment:
```suggestion
  auto bitmask =  array->buffers[validity_buffer_idx]; // not sure this is right
  auto out_col  = cudf::column_view(
                        data_type{type_id::STRING}
                        num_rows,
                        chars_column->view().head(),
                        static_cast<bitmask_type*>(bitmask),
                        input->null_count,
                        {offsets_column->view()});

  return std::make_unique<column>(
               cudf::detail::slice(out_col,
                                   static_cast<size_type>(input->offset),
                                   static_cast<size_type>(input->offset + input->length),
                                   stream),
               stream,
               mr);
```
Create a view to prevent making two columns when `offset!=0`
I think that something similar could be done with the offsets column since we only need the view to build the strings' column view.